oxd-server/src/main/java/org/gluu/oxd/server/Utils.java

```diff
@@ -97,4 +94,18 @@ public static boolean isTrue(Boolean bool) {
     public static List<String> stringToList(String source) {
         return Arrays.asList(source.split("\\s+"));
     }
+
+    public static String mapAsStringWithEncodedValues(Map<String, String> p_map) {
+        if (p_map != null && p_map.size() != 0) {
+            return p_map.entrySet().stream().map(e -> {
+                try {
+                    return e.getKey() + "=" + encode(e.getValue());
+                } catch (UnsupportedEncodingException e1) {
+                    throw new RuntimeException(e1);
+                }
+            }).collect(Collectors.joining("&"));
+        } else {
+            return null;
```

Comment:
lets return empty string `""` to avoid hypothetical NPE in future code.

Reply:
Change done